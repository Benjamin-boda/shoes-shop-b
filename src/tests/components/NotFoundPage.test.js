import React from "react";
import NotFoundPage from "../../components/NotFoundPage";
import {render, cleanup} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router-dom';

afterEach(cleanup);

const { asFragment, getByText } = render(
    <MemoryRouter>
        <NotFoundPage />
    </MemoryRouter>)

test("should match the snapshot", () => {
    
    expect(asFragment(<NotFoundPage/>)).toMatchSnapshot()
})

test("should contain this text", () => {
    const { getByText } = render(
        <MemoryRouter>
            <NotFoundPage />
        </MemoryRouter>)

    expect(getByText("404!")).toBeTruthy()

    expect(getByText("Go home")).toBeTruthy()
})