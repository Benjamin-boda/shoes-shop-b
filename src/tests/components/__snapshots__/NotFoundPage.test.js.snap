// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    404! 
    <a
      href="/"
    >
      Go home
    </a>
  </div>
</DocumentFragment>
`;
